src/main/java/appeng/recipes/factories/recipes/PartRecipeFactory.java

```diff
@@ -72,6 +72,48 @@ else if( type.contains( "shapeless" ) )
 		}
 	}
 
+	public static ItemStack getResult( JsonObject json, JsonContext context )
+	{
+		JsonObject resultObject = JsonUtils.getJsonObject( json, "result" );
+
+		if( resultObject.has( "part" ) )
+		{
+			return getPart( resultObject );
+		}
+		else if( resultObject.has( "item" ) )
+		{
+			return CraftingHelper.getItemStack( resultObject, context );
+		}
+		else
+		{
+			throw new JsonSyntaxException( "Result has no 'part' or 'item' property." );
+		}
+	}
+
+	private static ItemStack getPart( JsonObject resultObject )
+	{
+		String ingredient = JsonUtils.getString( resultObject, "part" );
+		Object result = (Object) Api.INSTANCE.registries().recipes().resolveItem( AppEng.MOD_ID, ingredient );
+		if( result instanceof ResolverResult )
+		{
+			ResolverResult resolverResult = (ResolverResult) result;
+
+			Item item = Item.getByNameOrId( AppEng.MOD_ID + ":" + resolverResult.itemName );
+
+			if( item == null )
+			{
+				AELog.warn( "item was null for " + resolverResult.itemName + " ( " + ingredient + " )!" );
+				throw new JsonSyntaxException( "Got a null item for " + resolverResult.itemName + " ( " + ingredient + " ). This should never happen!" );
+			}
+
+			return new ItemStack( item, JsonUtils.getInt( resultObject, "count", 1 ), resolverResult.damageValue, resolverResult.compound );
+		}
+		else
+		{
+			throw new JsonSyntaxException( "Couldn't find the resulting item in AE. This means AE was provided a recipe that it shouldn't be handling.\n" + "Was looking for : '" + ingredient + "'." );
+		}
+	}
+
 	// Copied from ShapedOreRecipe.java, modified a bit.
 	private static ShapedOreRecipe shapedFactory( JsonContext context, JsonObject json )
```

Comment:
![CRITICAL](https://sonarsource.github.io/sonar-github/severity-critical.png 'Severity: CRITICAL') The Cyclomatic Complexity of this method "shapedFactory" is 13 which is greater than 10 authorized. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.orod.org/coding_rules#rule_key=squid%3AMethodCyclomaticComplexity)